Dockerfile

```diff
@@ -0,0 +1,49 @@
+# syntax = docker/dockerfile:1
+ARG NODE_VERSION=22.9.0
+FROM node:${NODE_VERSION}-slim AS base
+ENV NODE_VERSION="$NODE_VERSION"
+
+
+LABEL fly_launch_runtime="Next.js"
+
+# Next.js app lives here
+WORKDIR /app
+
+# Install pnpm
+ARG PNPM_VERSION=9.12.0
+RUN npm install -g pnpm@$PNPM_VERSION
+
+
+FROM base AS build
+
+# Set these via `docker run -e APP_DEBUG=debug ...`
+# OR set them in your deployed containers environment secrets
+# these are needed for pnpm install schema:generate
+ENV APL='file'
+ENV APP_DEBUG='debug'
+ENV SECRET_KEY='test-see-comment-above'
+# do not replace these here, set these in the deployed environment, or docker run -e ...
+# ENV TEST_SALEOR_API_URL=''
+# ENV UPSTASH_TOKEN=''
+# ENV UPSTASH_URL=''
+
+ENV NODE_ENV="production"
+
+WORKDIR /app
+
+COPY package.json pnpm-lock.yaml ./
+RUN pnpm install --frozen-lockfile --prod=false
```

Comment:
From my understanding, at build time, the `SECRET_KEY` is only needed for `pnpm install`, if so, we should only set it for that step, like so:

```patch
--- Dockerfile	2024-10-09 10:31:45
+++ Dockerfile	2024-10-09 10:32:14
@@ -21,7 +21,6 @@
 # these are needed for pnpm install schema:generate
 ENV APL='file'
 ENV APP_DEBUG='debug'
-ENV SECRET_KEY='test-see-comment-above'
 # do not replace these here, set these in the deployed environment, or docker run -e ...
 # ENV TEST_SALEOR_API_URL=''
 # ENV UPSTASH_TOKEN=''
@@ -32,7 +31,7 @@
 WORKDIR /app

 COPY package.json pnpm-lock.yaml ./
-RUN pnpm install --frozen-lockfile --prod=false
+RUN SECRET_KEY=dummy pnpm install --frozen-lockfile --prod=false

 COPY . .
 RUN pnpm run build
```

My concern with the current `Dockerfile` is that `SECRET_KEY` is still set in the final image which causes the `Dockerfile` to have an insecure default value which could potentially be dangerous for the users of that image (as they may forget or not know they need to override the secret thus nullifying the security benefits or protections of `SECRET_KEY`).

For example, with this minimal `Dockerfile` (does the same as this PR):

```dockerfile
FROM busybox AS base

ENV SECRET_KEY=dummy

FROM base

ENTRYPOINT ["bash", "-i"]
```

We notice that `SECRET_KEY` is still set at runtime:

```bash
$ docker build -t test . && docker run -ti --rm --entrypoint env test
[...]
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=9e084b5acc66
TERM=xterm
SECRET_KEY=dummy  # <-- dangerous!
HOME=/root
```

Reply:
Ah yep, Thank you.

Pushed the change.

```
05:12:55 jm@pop-os saleor-app-payment-stripe 
±|canary ✗|→   docker build -t stripe2 --progress=plain . --no-cache
```
... some time later ...

````
05:12:35 jm@pop-os saleor-app-payment-stripe
 ±|canary ✗|→ docker run -it --rm stripe2 printenv

NODE_VERSION=22.9.0
HOSTNAME=66235543ded6
YARN_VERSION=1.22.22
HOME=/root
TERM=xterm
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/app
```